packages/common/src/repository/activeSubmissionRepository.ts

```diff
@@ -108,14 +109,17 @@ const repository = (dependencies: Dependencies) => {
 		 * @param {any} newData Set fields to update
 		 * @returns An updated record
 		 */
-		update: async (submissionId: number, newData: Partial<Submission>): Promise<Submission> => {
+		update: async (submissionId: number, newData: Partial<Submission>): Promise<Submission | null> => {
 			try {
-				const updated = await db
+				const resultUpdate = await db
 					.update(submissions)
 					.set({ ...newData, updatedAt: new Date() })
 					.where(eq(submissions.id, submissionId))
 					.returning();
-				return updated[0];
+				if (isArrayWithValues(resultUpdate)) {
+					return resultUpdate[0];
+				}
```

Comment:
If you're updating the function signature to allow `undefined` as a return type, then you can just leave this as `return updated[0];`

Reply:
the benefit I see having the `undefined` in the function's return type is that it indicates the caller to validate the value accordingly for undefined values.